e2e/validation.go

```diff
@@ -0,0 +1,179 @@
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: lvmVolumeGroupName, Namespace: InstallNamespace}, &lvmVG)
+		if err != nil {
+			debug("LVM vg: %v", err)
+			return false
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
+
+	debug("VG found\n")
+	return nil
+}
+
+// function to validate storage class.
+func ValidateStorageClass() error {
+	sc := storagev1.StorageClass{}
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: storageClassName}, &sc)
+		if err != nil {
+			debug("Storage class: %v", err)
+			return false
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
+
+	debug("SC found\n")
+	return nil
+}
+
+// function to validate CSI Driver.
+func ValidateCSIDriver() error {
+	cd := storagev1.CSIDriver{}
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: csiDriverName, Namespace: InstallNamespace}, &cd)
+		if err != nil {
+			debug("CSI Driver: %v", err)
+			return false
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
+
+	debug("CSI Driver found\n")
+	return nil
+}
+
+// function to validate TopoLVM node.
+func ValidateTopolvmNode() error {
+	ds := appsv1.DaemonSet{}
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: topolvmNodeDaemonSetName, Namespace: InstallNamespace}, &ds)
+		if err != nil {
+			debug("TopoLVM Node: %v", err)
+			return false
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
+	debug("TopoLVM node found\n")
+
+	/*
+		// checking for the ready status
+		Eventually(func() bool {
+			return ds.Status.DesiredNumberScheduled == ds.Status.NumberReady
+		}, timeout, interval).Should(BeTrue())
+		debug("Status is ready\n")*/
+
+	return nil
+}
+
+// function to validate vg manager resource.
+func ValidateVGManager() error {
+	ds := appsv1.DaemonSet{}
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: vgManagerDaemonsetName, Namespace: InstallNamespace}, &ds)
+		if err != nil {
+			debug("VG Manager: %v", err)
+			return false
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
+	debug("VG manager found\n")
+
+	return nil
+}
+
+// function to validate TopoLVM Deployment.
+func ValidateTopolvmController() error {
+	dep := appsv1.Deployment{}
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: topolvmCtrlDeploymentName, Namespace: InstallNamespace}, &dep)
+		if err != nil {
+			debug("TopoLVM Controller: %v", err)
+			return false
+		}
+		return true
+	}, timeout, interval).Should(BeTrue())
+
+	debug("depoloyment found\n")
+	return nil
+}
+
+// Validate all the resources created by LVMO.
+func ValidateResources() error {
+
+	// Validate Storage class
+	err := ValidateStorageClass()
```

Comment:
Change the order here to match the order in the LVMCluster reconcile  function:

1. CSI Driver
2. Topolvm controller
3. vg manager
4. LVMVolumeGroup
5. topolvm node
6. Storageclass


```

	// Validate CSI Driver
	err := ValidateCSIDriver()
	if err != nil {
		return err
	}

	//Validate TopoLVM Controller
	err = ValidateTopolvmController()
	if err != nil {
		return err
	}

	// Validate VG Manager Daemonset
	err = ValidateVGManager()
	if err != nil {
		return err
	}
	// Validate LVMVg
	err = ValidateLVMvg()
	if err != nil {
		return err
	}

	// Validate Topolvm node
	err = ValidateTopolvmNode()
	if err != nil {
		return err
	}
	// Validate Storage class
	err = ValidateStorageClass()
	if err != nil {
		return err
	}
```